cli/azd/pkg/infra/scope.go

```diff
@@ -19,7 +19,7 @@ type Scope interface {
 	// Gets the url to check deployment progress
 	DeploymentUrl() string
 	// Deploy a given template with a set of parameters.
-	Deploy(ctx context.Context, templatePath string, parametersPath string) error
+	Deploy(ctx context.Context, template string, parametersPath string) error
```

Comment:
The template is a file path at this point, right? Not a template string.  Why are we making this change?

Reply:
To prevent confusion, I wonder if we should do something like:

// ArmTemplate is a JSON encoded ARM template.
type ArmTemplate string

And then use `ArmTemplate` as the type of the parameter.  It might make it a little clearer as to what is going on here.

Reply:
added `ArmTemplate`.  I had to add it to a package that is common to all the involved packages (infra, provisioning and azcli).  I didn't want to define it 3 times (once for every package), and I don't want to make those packages to depend on each other (most likely a cycle)